### ISSUE TITLE
Fix false positives for functional psuedo-classes in selector-max-specificity

### ISSUE THREAD

**Issue by alexander-schranz** [stylelint/stylelint#3681]
> Clearly describe the bug

The selector max specificity errors when `nth-last-of-type()` or similiar selectors have a value e.g. `nth-last-of-type(2)` the specificity library returns correctly 0,0,3,0. But stylelint fails

> Which rule, if any, is the bug related to?

`selector-max-specificity`

> What CSS is needed to reproduce the bug?

```scss
.navigation__sub-list {
    // This code errors: 
    .navigation__item:nth-last-of-type(-n+2) > & {
         // code
    }

    // This doesnt: 
    .navigation__item:last-child > & {
         // code
    }
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "selector-max-specificity": "0,3,0"
  }
}
```

> Which version of stylelint are you using?

`9.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint css/*.scss css/**/*.scss`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS nested properties.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
Expected ".navigation__item:nth-last-of-type(-n+2) > .navigation__sub-list" to have a      selector-max-specificity
          specificity no more than "0,3,0"
```

> Additional:

```shell
node_modules/specificity/bin/specificity ".navigation__item:nth-last-of-type(-n+2) > .navigation__sub-list"
```

Returns correctly 0,0,3,0 which is the accepted specificity

**Comment by jeddy3** [comment]
@alexander-schranz Thanks for the report and for using the template.

I can reproduce this bug. We must be doing something with the selector before passing it along to the `specificity` library. I'll label as `help wanted` and bug.
